ws-client/src/manager.rs

```diff
@@ -34,32 +37,57 @@ pub enum RequestStatus {
 	Invalid,
 }
 
-type PendingCallOneshot = oneshot::Sender<Result<JsonValue, Error>>;
+type PendingCallOneshot = Option<oneshot::Sender<Result<JsonValue, Error>>>;
 type PendingSubscriptionOneshot = oneshot::Sender<Result<(mpsc::Receiver<JsonValue>, SubscriptionId), Error>>;
 type SubscriptionSink = mpsc::Sender<JsonValue>;
 type UnsubscribeMethod = String;
 type RequestId = u64;
 
-#[derive(Debug, Default)]
+/// Default max number of requests that are allowed in the `RequestManager`
+const REQUEST_MANAGER_DEFAULT_MAX_SIZE: u64 = 65535;
+
+#[derive(Debug)]
 /// Manages and monitors JSONRPC v2 method calls and subscriptions.
 pub struct RequestManager {
+	/// Vacant requestIDs.
+	free_slots: VecDeque<RequestId>,
 	/// List of requests that are waiting for a response from the server.
-	// NOTE: FnvHashMap is used here because RequestId is not under the caller's control and is known to be a short key (u64).
+	// NOTE: FnvHashMap is used here because RequestId is not under the caller's control and is known to be a short key.
 	requests: FnvHashMap<RequestId, Kind>,
 	/// Reverse lookup, to find a request ID in constant time by `subscription ID` instead of looking through all requests.
 	subscriptions: HashMap<SubscriptionId, RequestId>,
 }
 
+impl Default for RequestManager {
+	fn default() -> Self {
```

Comment:
this was super stupid because `new` calls `default::default` recursively and ends up with a stack overflow ^^